### PR TITLE
Add git rebase

### PR DESCRIPTION
**Pull request by xavierchow**

### CODE REVIEW

**Comment by makara** [comment]
Thanks. Let me rephrase some of it.

File: git/rebase.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+- You usually have to do `git stash`, `git pull --rebase`, `git stash pop` when you have local changes.
+- Alternatively, use Github Desktop, which does rebase with the "sync" automatically.
+
+### When you work on something
+
+- and you have a few commits in a new branch
+- be it pushed to a remote or not
+    - However know that if it's already pushed, you'll need to push it again with "force" after a rebase, ie. `git push -f`.
+
+A few scenarios can happen from here:
```
Comment (xavierchow) [inline git/rebase.md]: I want the scenarios to be more clear.

File: git/rebase.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+1. You want to change (edit, remove, or merge etc) some of the commits,
+    - Use `git rebase -i`, for example `git rebase -i master`, `git rebase -i HEAD~4` etc
```
Comment (xavierchow) [inline git/rebase.md]: Don't raise an example with `git rebase -i master`, it's confusing.

File: git/rebase.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+2. The base branch (often `master`) was changed and you need to include the change.
+    - Simply `git rebase [branch]`
```
Comment (xavierchow) [inline git/rebase.md]: For your case above, it should be switch to your branch by git check,
then git rebase master.

**Comment by makara** [comment]
We can improve later. Merging.
